pkg/kube/workload_ready_test.go

```diff
@@ -0,0 +1,135 @@
+package kube
+
+import (
+	"context"
+	"time"
+
+	. "gopkg.in/check.v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+type WorkloadReadySuite struct{}
+
+var _ = Suite(&WorkloadReadySuite{})
+
+type cliParams struct {
+	name                    string
+	namespace               string
+	stateful                bool
+	specReplicas            int32
+	statusReplicas          int32
+	statusUpdatedReplicas   int32
+	statusAvailableReplicas int32
+	observedGeneration      int64
+	podStatus               v1.PodPhase
+}
+
+func (s *WorkloadReadySuite) TestWaitOnStatefulSetReady(c *C) {
+	cp := cliParams{"ss", "default", true, 1, 1, 1, 1, 2, "Running"}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second/4)
+	defer cancel()
+	err := WaitOnStatefulSetReady(ctx, getCli(cp), cp.namespace, cp.name)
+	c.Assert(err, IsNil)
+}
+
+func (s *WorkloadReadySuite) TestStatefulSetReady(c *C) {
+	cp := cliParams{"ss", "default", true, 1, 1, 1, 1, 2, "Running"}
+	ctx := context.Background()
+	ready, status, err := StatefulSetReady(ctx, getCli(cp), cp.namespace, cp.name)
+	c.Assert(status, DeepEquals, "")
+	c.Assert(ready, DeepEquals, true)
+	c.Assert(err, IsNil)
+}
+
+func (s *WorkloadReadySuite) TestWaitOnDeploymentReady(c *C) {
+	cp := cliParams{"dep", "default", false, 5, 1, 1, 1, 2, "Running"}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second/4)
+	defer cancel()
+	err := WaitOnDeploymentReady(ctx, getCli(cp), cp.namespace, cp.name)
+	c.Assert(err, IsNil)
```

Comment:
@stevejoachim Looks like this is failing CI
```
Specified 5 replicas and only have 1: Context done while polling: context deadline exceeded ("Specified 5 replicas and only have 1: Context done while polling: context deadline exceeded")
```

Reply:
Thanks for catching that error. I just updated the PR.
In general, the tests were not doing anything useful as written. I was just using them during development to force the different error states.
To make them more useful, we could run a table-driven test that checks the different error messages. For an example, I rewrote `TestWaitOnDeploymentReady` this way. Might be overkill though. What do you think?

Reply:
The test looks good. You could maybe add cases for the remaining errors in the function and a case for nil error.

Reply:
Thanks for reviewing. I'll go ahead and add those additional tests.